gettsim/taxes/favorability_check.py

```diff
@@ -1,59 +1,152 @@
+"""
+This module contains the 'Higher-Yield Test':
+It compares the tax burden that results from various definitions of the tax base.
+Most importantly, it compares the tax burden without applying the child
+allowance (kein_kind_freib) AND receiving child benefit with the tax burden including
+the child allowance (kind_freib), but without child benefit. The most beneficial
+(for the household) is chosen. If child allowance is claimed, kindergeld is
+set to zero. A similar check applies to whether it is more profitable to
+tax capital incomes with the standard 25% rate or to include it in the tariff.
+"""
+import copy
+
 import numpy as np
+import pandas as pd
+
+
+def _beantrage_kind_freib_tu(
+    _st_kein_kind_freib_tu, _kindergeld_m_tu_basis, _st_kind_freib_tu
+):
+    """
+
+    Parameters
+    ----------
+    _st_kein_kind_freib_tu
+    _kindergeld_m_tu_basis
+    _st_kind_freib_tu
+
+    Returns
+    -------
 
+    """
+    st_kein_kind_freib = _st_kein_kind_freib_tu - 12 * _kindergeld_m_tu_basis
+    out = st_kein_kind_freib > _st_kind_freib_tu
+    return out.rename("_beantrage_kind_freib_tu")
+
+
+def _eink_st_m_tu_bis_1996(_st_kind_freib_tu):
+    """
+    Until 1996 individuals could claim child allowance and recieve child benefit.
+    Therefore the tax burden is allways smaller.
+    Parameters
+    ----------
+    _st_kind_freib_tu
+
+    Returns
+    -------
 
-def favorability_check(tax_unit, params):
-    """ 'Higher-Yield Test'
-        compares the tax burden that results from various definitions of the tax base
-        Most importantly, it compares the tax burden without applying the child
-        allowance (_nokfb) AND receiving child benefit with the tax burden including
-        the child allowance (_kfb), but without child benefit. The most beneficial (
-        for the household) is chosen. If child allowance is claimed, kindergeld is
-        set to zero. A similar check applies to whether it is more profitable to
-        tax capital incomes with the standard 25% rate or to include it in the tariff.
-    """
-    tax_unit["kindergeld_m"] = tax_unit["kindergeld_m_basis"]
-    tax_unit["kindergeld_m_tu"] = tax_unit["kindergeld_m_tu_basis"]
-    # get the maximum income
-    max_inc = get_max_inc(tax_unit, params)
-    # relevant incometax
-    tax_unit.loc[:, "eink_st_m_tu"] = 0
-    # Income Tax in monthly terms! And write only to parents
-    tax_unit.loc[~tax_unit["kind"], "eink_st_m_tu"] = (
-        tax_unit["_st_" + max_inc + "_tu"] / 12
+    """
+    out = copy.deepcopy(_st_kind_freib_tu)
+    return out.rename("eink_st_m_tu")
```

Comment:
I think we do not need a copy. We will never change a variable in-place and this will just pass the reference which saves memory. If we ever want to ensure that in-place operations are not possible, we should do copies of the inputs before every function is called in the dag.

```suggestion
    return _st_kind_freib_tu
```